frontend/public/components/pod.tsx

```diff
@@ -121,45 +287,104 @@ const PodTableRow = connect<PodTableRowPropsFromState, null, PodTableRowProps>(p
     style,
     metrics,
     showNodes,
+    selectedColumns,
   }: PodTableRowProps & PodTableRowPropsFromState) => {
-    const { name, namespace, creationTimestamp } = pod.metadata;
+    const { name, namespace, creationTimestamp, labels } = pod.metadata;
     const { readyCount, totalContainers } = podReadiness(pod);
     const phase = podPhase(pod);
     const restarts = podRestarts(pod);
     const bytes: number = _.get(metrics, ['memory', namespace, name]);
     const cores: number = _.get(metrics, ['cpu', namespace, name]);
+    const columns = new Set(
+      selectedColumns?.get(columnManagementID) || getSelectedColumns(showNodes),
+    );
     return (
       <TableRow id={pod.metadata.uid} index={index} trKey={rowKey} style={style}>
-        <TableData className={tableColumnClasses[0]}>
+        <TableData className={podColumnInfo.name.classes}>
           <ResourceLink kind={kind} name={name} namespace={namespace} />
         </TableData>
-        <TableData className={classNames(tableColumnClasses[1], 'co-break-word')}>
+        <TableData
+          className={classNames(podColumnInfo.namespace.classes, 'co-break-word')}
+          columns={columns}
```

Comment:
It's too bad we have to pass `columns` through to every single table cell. Is there a way to avoid that? Can `TableData` connect directly to Redux? Or maybe through `useContext`?